chrome/content/zotero/components/virtualized-table.jsx

```diff
@@ -0,0 +1,1350 @@
+/*
+	***** BEGIN LICENSE BLOCK *****
+	
+	Copyright © 2019 Center for History and New Media
+					George Mason University, Fairfax, Virginia, USA
+					http://zotero.org
+	
+	This file is part of Zotero.
+	
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+	
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero.	If not, see <http://www.gnu.org/licenses/>.
+	
+	***** END LICENSE BLOCK *****
+*/
+
+'use strict';
+
+const React = require('react');
+const PropTypes = require('prop-types');
+const cx = require('classnames');
+const JSWindow = require('./js-window');
+const Draggable = require('./draggable');
+const { injectIntl } = require('react-intl');
+const { IconDownChevron, getDomElement } = require('components/icons');
+
+const RESIZER_WIDTH = 5; // px
+
+const noop = () => 0;
+
+/**
+ * Somewhat corresponds to nsITreeSelection
+ */
+class TreeSelection {
+	constructor(tree) {
+		this._tree = tree;
+		Object.assign(this, {
+			pivot: 0,
+			_focused: 0,
+			selected: new Set([]),
+			_selectEventsSuppressed: false
+		});
+	}
+
+	isSelected(index) {
+		index = Math.max(0, index);
+		return this.selected.has(index);
+	}
+
+	toggleSelect(index, shouldDebounce) {
+		index = Math.max(0, index);
+		if (this.selected.has(index)) {
+			this.selected.delete(index);
+		}
+		else {
+			this.selected.add(index);
+		}
+
+		if (this.selectEventsSuppressed) return;
+
+		if (this._tree.invalidate) {
+			this._tree.invalidateRow(index);
+		}
+		this.pivot = index;
+		this._focused = index;
+		this._updateTree(shouldDebounce);
+	}
+
+	clearSelection() {
+		this.selected = new Set();
+		if (this._tree.invalidate) {
+			this._tree.invalidate();
+		}
+		this._updateTree();
+	}
+
+	/**
+	 * @param index
+	 * @returns {boolean} False if nothing to select and select handlers won't be called
```

Comment:
- Comment format here and elsewhere is missing the variable type
- Missing `shouldDebounce` explanation, here and elsewhere